core/src/main/java/io/parsingdata/metal/data/DataExpressionSource.java

```diff
@@ -45,17 +48,17 @@ public DataExpressionSource(final ValueExpression dataExpression, final int inde
     protected byte[] getData(final long offset, final int size) throws IOException {
         final ImmutableList<Optional<Value>> results = dataExpression.eval(graph, encoding);
         if (results.size <= index) { throw new IllegalStateException("ValueExpression dataExpression yields " + results.size + " result(s) (expected at least " + (index + 1) + ")."); }
-        final byte[] inputData = getValueAtIndex(results, index, 0).get().getValue();
+        final byte[] inputData = getValueAtIndex(results, index, 0).computeResult().get().getValue();
```

Comment:
With regard to methods: that's a good style. With private methods I tend to just immediately call the lowest level version :-)

With regard to final question: I think that's checked in the line above it?